### PR TITLE
Add Keycloak CR support for Tracing options

### LINKED ISSUE #32092: OTEL: Add Keycloak CR support for Tracing options
### Description

Tracing options are now part of Keycloak distribution, and we should also provide support for them in Keycloak CR. As all tracing capabilities are still in preview mode, the timeline might be unclear, as providing preview options in CR would probably not be the best way. 

### Motivation

Users should be able to set up tracing via Keycloak CR

### PR DESCRIPTION
**Pull request by mabartos**
- Closes #32092

Waiting for #35756 to be merged first

### CODE REVIEW

**Comment by mabartos** [comment]
@keycloak/cloud-native-maintainers @ahus1 Could you please check it? Thanks!

File: docs/guides/observability/tracing.adoc [hunk 1]
```diff
@@ -162,8 +162,41 @@ For more information, see the https://www.w3.org/TR/trace-context/#security-cons
...
+  tracing:
+    enabled: true                             # default 'false'
+    endpoint: http://my-tracing:4317          # default 'http://localhost:4317'
+    serviceName: my-best-keycloak             # default 'keycloak'
+    protocol: http/protobuf                   # default 'grpc'
+    samplerType: parentbased_traceidratio     # default 'traceidratio'
+    samplerRatio: 0.01                        # default '1'
+    compression: gzip                         # default 'none'
+    resourceAttributes:
+      service.namespace: keycloak-namespace
```
Comment (ahus1) [inline docs/guides/observability/tracing.adoc:188]: Nitpick: As we already include the namespace in `k8s.namespace.name` as described below, using `service.namespace` here might guide people to a situation where they have both in their traces. 

About `serviceName` - isn't the default the name of the CR, in this case `example-kc`?

I would shorten this to those parameters you would usually set, therefore skipping `protocol` and `compression`, and possibly `serviceName` as well.

File: docs/guides/observability/tracing.adoc [hunk 2]
```diff
@@ -162,8 +162,41 @@ For more information, see the https://www.w3.org/TR/trace-context/#security-cons
...
+  additionalOptions:
+    - name: tracing-jdbc-enabled
+      value: false                            # default 'true'
+----
+
+These fields should reflect 1:1 association with `tracing-*` options that contain more information.
+
+NOTE: The `tracing-jdbc-enabled` is not promoted as a first-class citizen as it might not be well managed in the future, so it needs to be set via the `additionalOptions` field.
+
+=== Filter traces based on k8s attributes
```
Comment (ahus1) [inline docs/guides/observability/tracing.adoc:199]:
```suggestion
=== Filter traces based on Kubernetes attributes
```

**Review by ahus1 — CHANGES_REQUESTED** [review]
Thank you for this PR. See below for some nitpicks around the docs.

File: docs/guides/observability/tracing.adoc (hunk 1 above)
Comment (mabartos) [reply docs/guides/observability/tracing.adoc:188]: > Nitpick: As we already include the namespace in k8s.namespace.name as described below, using service.namespace here might guide people to a situation where they have both in their traces.

> About serviceName - isn't the default the name of the CR, in this case example-kc?

> I would shorten this to those parameters you would usually set, therefore skipping protocol and compression, and possibly serviceName as well.

Agree, thanks!

File: docs/guides/observability/tracing.adoc [hunk 3]
```diff
@@ -162,8 +162,37 @@ For more information, see the https://www.w3.org/TR/trace-context/#security-cons
 == Tracing in Kubernetes environment
 When the tracing is enabled when using the {project_name} Operator, certain information about the deployment is propagated to the underlying containers.
 
-NOTE: There is no support for tracing configuration in {project_name} CR yet, so the `additionalOptions` can be used to the `tracing-enabled` property and other tracing options.
+=== Configuration via Keycloak CR
```
Comment (vmuzikar) [inline docs/guides/observability/tracing.adoc:165]: We're usually trying to keep any Operator related docs in the Operator guide. Since it's just a simple paragraph, IMHO we could leverage the [advance config guide](https://github.com/keycloak/keycloak/blob/main/docs/guides/operator/advanced-configuration.adoc) for it (we might want to split it in the future into several guides).

File: operator/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakDeploymentTest.java [hunk 4]
```diff
@@ -770,6 +773,86 @@ public void testApplyingResourcesDefaultValues() {
...
     }
 
+    @Test
+    public void testTracingSpec() {
+        var kc = getTestKeycloakDeployment(true);
+        kc.getSpec().setStartOptimized(false);
+
+        var featureSpec = new FeatureSpec();
+        featureSpec.setEnabledFeatures(List.of("opentelemetry"));
+        kc.getSpec().setFeatureSpec(featureSpec);
```
Comment (vmuzikar) [inline operator/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakDeploymentTest.java:783]: We shouldn't need this if #35756 gets merged first.

Comment (mabartos) [reply operator/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakDeploymentTest.java:783]: Yes, we will not need this after getting that merged.

File: docs/guides/observability/tracing.adoc (hunk 3 above)
Comment (mabartos) [reply docs/guides/observability/tracing.adoc:165]: I can include it in there in some simpler version. But I'd like to have it even present in more advanced form in the tracing guide. @vmuzikar does it work for you?

**Comment by mabartos** [comment]
@vmuzikar @ahus1 Comments should be addressed and it should be ready to be reviewed now. Thanks!

**Review by ahus1 — APPROVED** [review]
Thank you for the changes, LGTM

**Review by vmuzikar — APPROVED** [review]
LGTM, nice work, thanks!
